sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,258 @@ def named_poly(n, f, K, name, x, polys):
+
+        Returns
+        =======
+        list of, list of tuples
+            A list where each inner list represents a valid combination of factors.
+
+        Raises
+        ======
+        TypeError
+            If any element in polys is not a Poly instance
+        DomainError
+            If all polynomials don't share the same domain and generators
+
+        Example
+        =======
+        >>> from sympy import symbols, Poly, ZZ
+        >>> from sympy.polys.polytools import factor_system_poly
+        >>> a, b, x = symbols('a b x')
+        >>> factor_system_poly([Poly((a - 1)*(x - 2), x, domain=ZZ[a,b]), Poly((b - 3)*(x - 2), x, domain=ZZ[a,b])])
+        [[(Poly(x - 2, x, domain='ZZ[a,b]'), [Poly(a - 1, x, domain='ZZ[a]'), Poly(b - 3, x, domain='ZZ[b]')])]]
+
+        Notes
+        =====
+        - Returns [[]] if the input list is empty
+        - Returns [] if any polynomial has degree 0 and is non-zero
+        - Returns [] if any non-zero polynomial has no factors but a non-zero constant term
+        - The function finds minimal combinations of factors, meaning no combination
+          is a proper superset of another
+    """
+    if not polys:
+        return [[]]
+
+    if not all(isinstance(poly, Poly) for poly in polys):
+        raise TypeError("polys should be a list of Poly instances")
+
+    base_domain = polys[0].domain
+    base_gens = polys[0].gens
+
+    if not all(poly.domain == base_domain and poly.gens == base_gens for poly in polys[1:]):
+        raise DomainError("All polynomials must have the same domain and generators")
+
+    if any(poly.total_degree() == 0 and not poly.is_zero for poly in polys):
+        return []
+
+    factored_eqs = []
+    for poly in polys:
+        constant, factors = poly.factor_list()
+
+        if constant.is_zero and not factors:
+            continue
+
+        current_factors = []
+        if factors:
+            if constant.is_number and constant != 0:
+                current_factors.extend((Poly(factor, *base_gens), []) for factor, _ in factors)
+            else:
+                current_factors.extend((Poly(factor, *base_gens), [Poly(constant, *base_gens)])
+                                       for factor, _ in factors)
+        elif not constant.is_zero:
+            return []
+
+        if current_factors:
+            factored_eqs.append(current_factors)
+
+    if not factored_eqs:
+        return [[]]
+
+    return _combine_factor_systems_poly(factored_eqs)
+
+
+def _combine_factor_systems_poly(factored_eqs):
+    """ Helper function to combine factored polynomial
+        equations into minimal factor systems. """
+    factor_dict = {}
+    for eq in factored_eqs:
+        for factor, conditions in eq:
+            if factor not in factor_dict:
+                factor_dict[factor] = []
+            factor_dict[factor].extend(conditions)
+
+    eq_factors = [{factor for factor, _ in eq} for eq in factored_eqs]
+
+    def _factors_system(eqs):
```

Comment:
Move this function out of the body of the other function and give it a more meaningful name and a short docstring that explains what it does.